sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/DocumentHelper.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Runtime.CompilerServices;
+using Azure.Monitor.OpenTelemetry.Exporter.Internals;
+using Azure.Monitor.OpenTelemetry.LiveMetrics.Models;
+
+namespace Azure.Monitor.OpenTelemetry.LiveMetrics.Internals
+{
+    /// <summary>
+    /// Helper class used to convert Activity to the LiveMetrics document schema.
+    /// </summary>
+    internal static class DocumentHelper
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static RemoteDependency ConvertToRemoteDependency(Activity activity)
+        {
+            string urlFull = string.Empty, httpResponseStatusCode = string.Empty;
+
+            foreach (ref readonly var tag in activity.EnumerateTagObjects())
+            {
+                if (tag.Value == null)
+                {
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeUrlFull)
+                {
+                    urlFull = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeHttpResponseStatusCode)
+                {
+                    httpResponseStatusCode = tag.Value.ToString();
+                    continue;
+                }
+            }
+
+            RemoteDependency remoteDependencyDocumentIngress = new()
+            {
+                DocumentType = DocumentIngressDocumentType.RemoteDependency,
+                Name = activity.DisplayName,
+                CommandName = urlFull, // TODO: WHAT ABOUT DATABASES?
```

Comment:
How does AI SDK gets value for DB/messaging?

Reply:
Can we include all the types (HTTP/RPC/Messaging) here? I want to understand how the 'foreach' loop and additional variables contribute to the overhead. 

The value must come from 'RemoteDependency.Data'.

https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/RemoteDependencyData.cs

Reply:
abandoned this implementation. Switched to ActivityTagsProcessor from Exporter